frontend/src/metabase/components/form/CustomForm/types.ts

```diff
@@ -0,0 +1,93 @@
+import PropTypes from "prop-types";
+import {
+  FieldName,
+  DefaultFieldValue,
+  FieldValues,
+  FormFieldDefinition,
+  FormField,
+  FormObject,
+} from "metabase-types/forms";
+
+export interface BaseFormProps {
```

Comment:
As far as I understand, some if not most of these are part of `redux-form` right? It's quite confusing where these might be coming from. Maybe adding a comment or using types directly from `@types/redux-form` would make it more obvious? What do you thinK?

Reply:
@WiNloSt yeah, form types are pretty heavy as there are a few different ways to render a form using the `Form` container:

**Entity form ([example](https://github.com/metabase/metabase/blob/17aa073720964f210bd84406a73a5f7422783722/frontend/src/metabase/collections/containers/CollectionCreate.jsx#L45))**

```jsx
<Collection.Form collection={collection} />
```

**Field definitions ([example](https://github.com/metabase/metabase/blob/17aa073720964f210bd84406a73a5f7422783722/frontend/src/metabase/query_builder/components/DatasetEditor/DatasetFieldMetadataSidebar/DatasetFieldMetadataSidebar.jsx#L296))**

```jsx
<Form
  form={{
    fields: [ { name: "name" } ]
  }}
/>
```

**Inline fields ([example](https://github.com/metabase/metabase/blob/17aa073720964f210bd84406a73a5f7422783722/enterprise/frontend/src/metabase-enterprise/auth/components/SettingsSAMLForm.jsx#L48))**

```jsx
<Form>
  <FormField name="name">
  <FormField name="description" type="text">
  <FormField name="collection" type="collection_id">
<Form/>
```

As a result, Form's props are pretty bloated as they handle a few different APIs that can also be used simultaneously. . A lot of props are coming from `redux-form`, I wouldn't actually highlight them as we're going to remove it. I'm trying to replace it without changing the external API too much, so we don't need to patch every single form in Metabase. So the new Form component is going to follow the same props structure / API and so the types will stay like this for some time